notebooks/406-3D-pose-estimation-webcam/406-3D-pose-estimation.ipynb

```diff
@@ -0,0 +1,597 @@
+    "\n",
+    "            # inference start\n",
+    "            start_time = time.time()\n",
+    "            # get results\n",
+    "            inference_result = model_infer(scaled_img, stride)\n",
+    "            poses_3d, poses_2d = parse_poses(inference_result, 1, stride, fx, True)\n",
+    "            # inference stop\n",
+    "            stop_time = time.time()\n",
+    "            processing_times.append(stop_time - start_time)\n",
+    "            # use processing times from last 200 frames\n",
+    "            if len(processing_times) > 200:\n",
+    "                processing_times.popleft()\n",
+    "\n",
+    "            processing_time = np.mean(processing_times) * 1000\n",
+    "            fps = 1000 / processing_time\n",
+    "            cv2.putText(\n",
+    "                scaled_img,\n",
+    "                f\"Inference time: {processing_time:.1f}ms ({fps:.1f} FPS)\",\n",
+    "                (10, 30),\n",
+    "                cv2.FONT_HERSHEY_COMPLEX,\n",
+    "                0.7,\n",
+    "                (0, 0, 255),\n",
+    "                1,\n",
+    "                cv2.LINE_AA,\n",
+    "            )\n",
+    "\n",
+    "            if len(poses_3d) > 0:\n",
+    "                # From here, you can rotate the 3D point positions using the function \"draw_poses\",\n",
+    "                # or you can directly make the correct mapping below to properly display the object image on the screen\n",
+    "                poses_3d_copy = poses_3d.copy()\n",
+    "                x = poses_3d_copy[:, 0::4]\n",
+    "                y = poses_3d_copy[:, 1::4]\n",
+    "                z = poses_3d_copy[:, 2::4]\n",
+    "                poses_3d[:, 0::4], poses_3d[:, 1::4], poses_3d[:, 2::4] = (\n",
+    "                    -z + np.ones(poses_3d[:, 2::4].shape) * 200,\n",
+    "                    -y + np.ones(poses_3d[:, 2::4].shape) * 100,\n",
+    "                    -x,\n",
+    "                )\n",
+    "\n",
+    "                poses_3d = poses_3d.reshape(poses_3d.shape[0], 19, -1)[:, :, 0:3]\n",
+    "                people = skeleton(poses_3d=poses_3d)\n",
+    "\n",
+    "                try:\n",
+    "                    engine3D.scene_remove(skeleton_set)\n",
+    "                except Exception:\n",
+    "                    pass\n",
+    "\n",
+    "                engine3D.scene_add(people)\n",
+    "                skeleton_set = people\n",
+    "\n",
+    "                # draw 2D\n",
+    "                draw_poses(scaled_img, poses_2d)\n",
+    "\n",
+    "            else:\n",
+    "                try:\n",
+    "                    engine3D.scene_remove(skeleton_set)\n",
+    "                    skeleton_set = None\n",
+    "                except Exception:\n",
+    "                    pass\n",
+    "\n",
+    "            if use_popup:\n",
+    "                frame = cv2.resize(scaled_img, dsize=(frame.shape[1], frame.shape[0]))\n",
```

Comment:
Please don't change size of image

Reply:
Yes. You should change the image size, but it should be like the following:
1. You create a frame copy and resize it to fit the neural network input size (scaled image)
2. You predict on the scaled image 
3. You scale the results (not the image) back to fit the original frame size
4. You draw on the original frame

Otherwise, the final image is warped, what I can observe now.